src/homework/tic_tac_toe/tic_tac_toe_manager.cpp

```diff
@@ -1,13 +1,15 @@
 #include <string>
 #include <vector>
 #include <iostream>
+#include <memory>
 #include "tic_tac_toe_manager.h"
+using std::string;
 using namespace std;
 
-void TTTManager::save_game(TicTacToe b)
+void TTTManager::save_game(unique_ptr<TicTacToe> &b)
 {
-    games.push_back(b);
-    update_winner_count(b.get_winner());
+    update_winner_count(b->get_winner());
+    //Could not get push back to work. Main would always crash when the function was called.
```

Comment:
add:
games.push_back(std::move(b));

Reply:
For test case testing TicTacToeManager make sure to create a new unique_ptr<TicTacToe3> or unique_ptr<TicTacToe4> each time you add it to manager, see below:

	TTTManager manager;
	unique_ptr<TicTacToe> match = make_unique<TicTacToe4>();
	 

	int o, x, t;
	string player = "X";
	match->start_game(player);
	match->mark_board(1);
	match->mark_board(2);
	match->mark_board(5);
	match->mark_board(4);
	match->mark_board(9);
	match->mark_board(10);
	match->mark_board(13);
	REQUIRE(match->game_over()== true);
	manager.save_game(match);

	match = make_unique<TicTacToe4>();
	match->start_game("O");
	match->mark_board(1);
	match->mark_board(2);
	match->mark_board(5);
	match->mark_board(4);
	match->mark_board(9);
	match->mark_board(10);
	match->mark_board(13);
	REQUIRE(match->game_over()== true);
	manager.save_game(match);
	
	match = make_unique<TicTacToe4>();
	match->start_game(player);
	match->mark_board(1);
	match->mark_board(2);
	match->mark_board(3);
	match->mark_board(4);
	match->mark_board(5);
	match->mark_board(6);
	match->mark_board(7);
	match->mark_board(8);
	match->mark_board(10);
	match->mark_board(9);
	match->mark_board(12);
	match->mark_board(11);
	match->mark_board(13);
	match->mark_board(14);
	match->mark_board(15);
	match->mark_board(16);
	REQUIRE(match->game_over()== true);
	manager.save_game(match);
	
	manager.get_winner_total(o,x,t);
	REQUIRE(o == 1);
	REQUIRE(x == 1);
	REQUIRE(t == 1);